### PR TITLE
Use ci-helpers (from Astropy) for CI and enable AppVeyor

### PR DESCRIPTION
**Pull request by astrofrog**
Over at the Astropy project, we've developed [ci-helpers](https://github.com/astropy/ci-helpers), which is a set of well tested scripts to help with setting up conda environments for continuous integration. This is not at all astronomy specific and we are interested in getting other projects to adopt it too. This PR shows how you can use it in the Travis configuration, and also adds a configuration file for AppVeyor, in case you would like to turn that on. I've also enabled OSX testing on Travis.

[to see the benefits of using ci-helpers, note that this PR only adds a net 6 lines but that includes adding the appveyor configuration]

If you are interested in going ahead with this, try turning on AppVeyor and I'll force push to this branch again to trigger a build.

Note that https://github.com/spyder-ide/qtpy/pull/28 should be merged first, after which I'll rebase this.

### CODE REVIEW

**Comment by astrofrog** [comment]
This is ready for a preliminary review - although note that there may still be failures once/if AppVeyor is turned on.

**Comment by goanpeca** [comment]
@ccordoba12, appveyor?

File: .travis.yml [hunk 1]
```diff
@@ -1,71 +1,41 @@
-language: python
+language: c
```
Comment (ccordoba12) [inline .travis.yml]: Why this change in `language`?

File: .travis.yml [hunk 2]
```diff
@@ -1,71 +1,41 @@
...
+
 sudo: false
 
+os:
+  - linux
+  - osx
```
Comment (ccordoba12) [inline .travis.yml]: Nice to have tests in OS X!!

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+platform:
+    -x64
+
+install:
+    - "git clone git://github.com/astropy/ci-helpers.git"
+    - "powershell ci-helpers/appveyor/install-miniconda.ps1"
+    - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
+    - "activate test"
+
+# Not a .NET project, we build SunPy in the install step instead
```
Comment (ccordoba12) [inline appveyor.yml]: Please remove or rewrite this comment.

File: .travis.yml (hunk 1 above)
Comment (astrofrog) [inline .travis.yml]: We can only run MacOS X if language is set to c or obj-c. In any case, we don't need to worry about the language since we use conda to set everything up, so we don't need python pre-installed.

File: appveyor.yml (hunk 3 above)
Comment (astrofrog) [inline appveyor.yml]: Ok, I've done this locally - let me know once AppVeyor is enabled and I'll push the changes.

File: .travis.yml (hunk 1 above)
Comment (ccordoba12) [inline .travis.yml]: Ok, thanks for the explanation.

File: appveyor.yml (hunk 3 above)
Comment (ccordoba12) [inline appveyor.yml]: I just activated AppVeyor:

https://ci.appveyor.com/project/ccordoba12/qtpy

So please go ahead :-)

Comment (astrofrog) [inline appveyor.yml]: Thanks!

**Comment by astrofrog** [comment]
I'm temporarily disabling Travis while I debug appveyor

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      CMD_IN_ENV: "cmd /E:ON /V:ON /C .\\ci-helpers\\appveyor\\windows_sdk.cmd"
+      PYTHON_ARCH: "64" # needs to be set for CMD_IN_ENV to succeed. If a mix
+                        # of 32 bit and 64 bit builds are needed, move this
+                        # to the matrix section.
+
+  matrix:
+
+    - PYTHON_VERSION: "2.7"
+      CONDA_CHANNELS: "spyder-ide"
+      CONDA_DEPENDENCIES: "pytest qt5 pyqt5"
```
Comment (ccordoba12) [inline appveyor.yml]: Please comment this section. There are no 2.7 packages for PyQt5 yet.

File: appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    - PYTHON_VERSION: "2.7"
+      CONDA_DEPENDENCIES: "pytest qt pyqt"
+    - PYTHON_VERSION: "2.7"
+      CONDA_DEPENDENCIES: "pytest qt pyside"
+    - PYTHON_VERSION: "3.5"
+      CONDA_CHANNELS: "spyder-ide"
```
Comment (ccordoba12) [inline appveyor.yml]: This needs to be changed to `"msarahan"`:

https://anaconda.org/msarahan/qt5/files

**Comment by astrofrog** [comment]
For now, I've rebased this on top of #28 since it provides the first test that is needed for py.test to pass. But we should merge #28 first then I can rebase this again.

File: .travis.yml (hunk 1 above)
Comment (Nodd) [inline .travis.yml]: Maybe add a comment then, since it's not obvious ?

Comment (astrofrog) [inline .travis.yml]: Done!

**Comment by astrofrog** [comment]
@ccordoba12 - the PyQt5 install doesn't work on Windows, even with @msarahan's conda build - it complains about a missing ``icu`` package. Shall I disable PyQt5 entirely on AppVeyor for now?

File: .travis.yml (hunk 2 above)
Comment (goanpeca) [inline .travis.yml]: @ccordoba12 can we do the same for spyde then :-p ?

**Comment by ccordoba12** [comment]
Yes, please deactivate it. Official packages will be available next week anyway :-)

File: .travis.yml (hunk 2 above)
Comment (ccordoba12) [inline .travis.yml]: Some day my friend, some day.... ;-)

File: tests/test_main.py [hunk 6]
```diff
@@ -33,7 +34,12 @@ def assert_pyqt5():
     assert QtCore.QEvent is PyQt5.QtCore.QEvent
     assert QtGui.QPainter is PyQt5.QtGui.QPainter
     assert QtWidgets.QWidget is PyQt5.QtWidgets.QWidget
-    assert QtWebEngineWidgets.QWebEnginePage is PyQt5.QtWebKitWidgets.QWebPage
+    from PyQt5 import Qt
+    if LooseVersion(Qt.PYQT_VERSION_STR) < LooseVersion('5.6'):
```
Comment (ccordoba12) [inline tests/test_main.py]: This is not exactly right because WebKit can still be compiled and distributed with Qt 5.6.

In fact, the Qt 5.6 versions that we're building at Continuum for Python 2.7 and Python 3.4 on Windows will come with WebKit instead of WebEngine because WebEngine requires VS 2015.

That's why we added a constant called `WEBENGINE` to `QtWebEngineWidgets` in qtpy 1.0.2: to detect which Qt web backend is currently in use.

Please use that constant here :-)

Comment (astrofrog) [inline tests/test_main.py]: No problem - should be fixed now!

**Comment by astrofrog** [comment]
@ccordoba12 - I think this is good to go?

**Comment by ccordoba12** [comment]
Yep, thanks a lot for helping us with this!
